components/push/src/communications.rs

```diff
@@ -196,9 +199,12 @@ impl Connection for ConnectHttp {
             if requested.status == status_codes::CONFLICT {
                 return Err(AlreadyRegisteredError.into());
             }
-            return Err(
-                CommunicationError(format!("Unhandled client error {:?}", requested)).into(),
-            );
+            return Err(CommunicationError(format!(
+                "Unhandled client error {} : {:?}",
+                requested.status,
+                String::from_utf8_lossy(&requested.body)
```

Comment:
Hmm, is there any risk of PII in the body?

Reply:
Quite the opposite. Our push servers return [extended help info](https://autopush.readthedocs.io/en/latest/http.html?highlight=error#error-resp) whenever they hit an error.  This info was getting lost in how the message was being returned.